resolver/resolver.go

```diff
@@ -72,7 +72,8 @@ func SetDefaultScheme(scheme string) {
 	defaultScheme = scheme
```

Comment:
This also needs to set `internal.UserSetDefaultScheme=true`.

Reply:
I had initially tried that, but was getting an `import cycle ` error when building. I saw this note on top of the `SetDefaultScheme` method

```go
// NOTE: this function must only be called during initialization time (i.e. in
// an init() function)
func SetDefaultScheme(scheme string) {
```

which was why I put it in the `init` function inside of `clientconn.go`

Reply:
This is the cycle its detecting

```bash
go build google.golang.org/grpc/...
package google.golang.org/grpc
	imports google.golang.org/grpc/balancer
	imports google.golang.org/grpc/channelz
	imports google.golang.org/grpc/internal/channelz
	imports google.golang.org/grpc/internal
	imports google.golang.org/grpc/resolver
	imports google.golang.org/grpc/internal: import cycle not allowed
```

Reply:
`internal` should not be importing `resolver`.  Actually it shouldn't import anything else in the grpc tree, as part of its intent, but we don't need to fix all the offenses right away.

Can you move `Get/SetXDSHandshakeClusterName` into another package?  Just `internal/xds/xds.go` seems fine to me.